lib/saluki-components/src/sources/checks/check.rs

```diff
@@ -5,11 +5,16 @@ use saluki_error::GenericError;
 /// Check trait
 ///
 /// This trait allow us to have different check implementations.
+#[allow(dead_code)]
 pub trait Check {
     /// Run the check
     fn run(&self) -> Result<(), GenericError>;
     /// Get the interval of the check
-    fn interval(&self) -> Duration;
+    fn interval(&self) -> &Duration;
```

Comment:
`std::time::Duration` is `Copy`: there's almost never a good reason to pass a reference to it.

Reply:
Thanks for the info. TIL